rxbinding-support-v4/src/androidTest/java/com/jakewharton/rxbinding/support/v4/widget/RxSwipeRefreshLayoutTestActivity.java

```diff
@@ -18,15 +21,21 @@
   private final Runnable stopRefreshing = new Runnable() {
     @Override public void run() {
       swipeRefreshLayout.setRefreshing(false);
-      handler.postDelayed(this, 500);
     }
   };
 
   @Override protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
     swipeRefreshLayout = new SwipeRefreshLayout(this);
-    swipeRefreshLayout.setId(1);
+    swipeRefreshLayout.setId(R.id.swipe_refresh_layout);
+    swipeRefreshLayout.setOnTouchListener(new View.OnTouchListener() {
+      @Override public boolean onTouch(View v, MotionEvent event) {
+        handler.removeCallbacks(stopRefreshing);
```

Comment:
Yes, we could do it on `ACTION_UP` as well with shorter delay, that way we also get flexibility for swipe speeds. Want me to do it that way?